### PR TITLE
[14.0][FIX]product_supplierinfo_for_customer_sale: set explicit quantity=None

### PR DESCRIPTION
**Pull request by PicchiSeba**
explicitly assign `quantity=None`

requires https://github.com/OCA/product-attribute/pull/1506

### CODE REVIEW

File: product_supplierinfo_for_customer_sale/tests/test_product_supplierinfo_for_customer_sale.py [hunk 1]
```diff
@@ -79,7 +79,10 @@ def _create_pricelist_item(self, name, pricelist, product):
 
     def test_product_supplierinfo_for_customer_sale(self):
         so = self.env["sale.order"].create(
-            {"partner_id": self.customer.id, "pricelist_id": self.pricelist.id}
+            {
+                "partner_id": self.customer.id,
+                "pricelist_id": self.pricelist.id,
+            }
```
Comment (HekkiMelody) [inline product_supplierinfo_for_customer_sale/tests/test_product_supplierinfo_for_customer_sale.py]: **chore**: Please remove this diff

File: product_supplierinfo_for_customer_sale/models/sale_order_line.py [hunk 2]
```diff
@@ -52,11 +53,11 @@ def product_id_change(self):
                     "&",
                     ("product_id", "=", product.id),
                     ("applied_on", "=", "0_product_variant"),
-                ]
+                ],
```
Comment (HekkiMelody) [inline product_supplierinfo_for_customer_sale/models/sale_order_line.py]: **chore**: please remove this diff

**Review by HekkiMelody — APPROVED** [review]
Code review ok

minor code cleanup.

**Comment by francesco-ooops** [comment]
@pedrobaeza good for you? :)

**Comment by pedrobaeza** [comment]
I think this should depend also on `product_uom_qty` and pass such quantity to the method.

**Comment by HekkiMelody** [comment]
> I think this should depend also on `product_uom_qty` and pass such quantity to the method.

@pedrobaeza We tried this, it was also our first line of thinking, but the module (and the tests) explicitly assume that you should be able to get a customer code even if you don't respect the minimum quantity. 

For example, in the tests it creates supplierinfo with minimum quantity set to 15, and then expects to get the customer code even if the quantity is set to 1.

https://github.com/OCA/sale-workflow/blob/14.0/product_supplierinfo_for_customer_sale/tests/test_product_supplierinfo_for_customer_sale.py#L47-L57

I'm unsure if that's how it's supposed to work, so for now we restored the original functionality (I haven't had time to look deeply into how it works for the original supplierinfo in base Odoo). If you think we should change it we can adapt the tests.

**Comment by pedrobaeza** [comment]
OK, let's continue for now as it is:

/ocabot merge patch
